### PR TITLE
Consent reminder setup

### PR DESCRIPTION
**Pull request by VinayJariya**
Resolves JIRA Ticket: [PE-2419](https://thriveworks.atlassian.net/browse/PE-2419),  [PE-2420](https://thriveworks.atlassian.net/browse/PE-2420)

UXPin - Figma Link

Briefing
Cron Job for sending consent reminder to patients and summary to support.

Associated PRs
Subtask of  [PE-2244](https://thriveworks.atlassian.net/browse/PE-2244)

How to test:
Instructions for testing locally, including data setup steps, etc..

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const isOffline = process.env.IS_OFFLINE === 'true';
+
+const keys = [
+  'POSTGRES_HOST', 'POSTGRES_PORT', 'POSTGRES_DATABASE', 'POSTGRES_USER','POSTGRES_PASSWORD', 'POSTGRES_POOL_SIZE',
```
Comment (prakash-velotio) [inline config.js:4]: Use read replica here

File: config.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const isOffline = process.env.IS_OFFLINE === 'true';
```
Comment (akash-rajput) [inline config.js:1]: this file is duplicated across multiple repos. We can move this to a common package. Please add a TODO for now.

Comment (prakash-velotio) [reply config.js:1]: Yupp,
We also have multiple interfaces getting duplicated across repos, Lets create central repo in next sprint

File: index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+async function loadDBClient() {
+  pg_client = require("./pg-client");
+}
+
+exports.handler = async (event, context) => {
+  try {
+    console.log(`event params`, event);
+    console.log(`context params`, context);
+    const { ENV, HAS_READ_REPLICA } = event;
```
Comment (prakash-velotio) [inline index.js:19]: We enabled read replica for stage and prod and for dev and qa we can use same db values in read replica variable on ssm

File: index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            where a.start BETWEEN '${paymentCheckWindow.minus({ hours: 1 })}' AND '${paymentCheckWindow}'
+            group by a.id, u.id, pro.id, pra.id`
+
+           const appointments = await pg_client.query(query);
+
+           for (let appointment of appointments.rows) {
+             const patientsNotHavingConsents = []
+             for(let patient of appointment.patients){
+             const patientConsent = patient['patient_consents']
+             const bookAppointmentConsent = patientConsent.find(consent => {
```
Comment (prakash-velotio) [inline index.js:73]: can we move this part to query ? 
Fetch only those patients along with appt. who has not provided consent

File: index.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    appBaseUrl: getValue('APP_BASE_URL')
+  };
+};
+
+//Need to revisit
+function getSupportEmailPayload(appointment) {
+  return {
+    appBaseUrl: getValue('APP_BASE_URL')
+  };
+};
```
Comment (prakash-velotio) [inline index.js:135]: linter fix

File: mailgun.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    return result;
+  } catch (e) {
+    console.error(e, 'Error thrown by mailgun : ');
+  }
+}
+
+module.exports = {
+  sendEmail,
+};
```
Comment (prakash-velotio) [inline mailgun.js:37]: Same linter

**Review by prakash-velotio — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (akash-rajput) [inline index.js:73]:
```suggestion
             const isConsentProvided = patientConsent.find(consent => {
```

**Review by akash-rajput — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (VinayJariya) [reply index.js]: Query will get complex. Lets keep it here only

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.serverless
+.env
```
Comment (prakash-velotio) [inline .gitignore:3]: linter fix

File: .env.example [hunk 8]
```diff
@@ -0,0 +1,15 @@
+POSTGRES_HOST=<placeholder>
```
Comment (prakash-velotio) [inline .env.example:1]: Update these vars to match read replica naming convention

File: .env.example [hunk 9]
```diff
@@ -0,0 +1,15 @@
+POSTGRES_HOST_READ=<placeholder>
```
Comment (prakash-velotio) [inline .env.example:1]: @VinayJariya Are you sure these are correct names ?
I think they are different from what @akashjobanputra has added

File: config.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+async function loadConfig(env = "qa") {
+  console.log(`Loading config vars from ${isOffline ? "dotenv" : "SSM"} `);
+  if (!isOffline) {
+    const AWS = require("aws-sdk");
+
+    const ssm = new AWS.SSM();
+    const paramNames = keys.map((key) => `/${env}/${key}`);
+    const ssmResp = await ssm
+      .getParameters({ Names: paramNames, WithDecryption: true })
```
Comment (prakash-velotio) [inline config.js:30]: @VinayJariya Are you sure all the vars are getting pulled here ?
@akashjobanputra What is the fetch limit of vars ?

Comment (akash-rajput) [reply config.js:30]: 10

**Review by prakash-velotio — APPROVED** [review]

**Pull request merged**